src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java

```diff
@@ -164,71 +158,29 @@ public void onComplete() {
-
         private final Path file;
         private final OpenOption[] options;
-        @SuppressWarnings("removal")
-        private final AccessControlContext acc;
-        private final FilePermission[] filePermissions;
-        private final boolean isDefaultFS;
         private final CompletableFuture<Path> result = new MinimalFuture<>();
 
         private final AtomicBoolean subscribed = new AtomicBoolean();
         private volatile Flow.Subscription subscription;
         private volatile FileChannel out;
 
-        private static final String pathForSecurityCheck(Path path) {
-            return path.toFile().getPath();
-        }
-
         /**
          * Factory for creating PathSubscriber.
-         *
-         * Permission checks are performed here before construction of the
-         * PathSubscriber. Permission checking and construction are deliberately
-         * and tightly co-located.
          */
         public static PathSubscriber create(Path file,
                                             List<OpenOption> options) {
-            @SuppressWarnings("removal")
-            SecurityManager sm = System.getSecurityManager();
-            FilePermission filePermission = null;
-            if (sm != null) {
-                try {
-                    String fn = pathForSecurityCheck(file);
-                    FilePermission writePermission = new FilePermission(fn, "write");
-                    sm.checkPermission(writePermission);
-                    filePermission = writePermission;
-                } catch (UnsupportedOperationException ignored) {
-                    // path not associated with the default file system provider
-                }
-            }
-
-            assert filePermission == null || filePermission.getActions().equals("write");
-            @SuppressWarnings("removal")
-            AccessControlContext acc = sm != null ? AccessController.getContext() : null;
-            return new PathSubscriber(file, options, acc, filePermission);
+            return new PathSubscriber(file, options);
         }
 
         // pp so handler implementations in the same package can construct
         /*package-private*/ PathSubscriber(Path file,
-                                           List<OpenOption> options,
-                                           @SuppressWarnings("removal") AccessControlContext acc,
-                                           FilePermission... filePermissions) {
+                                           List<OpenOption> options) {
             this.file = file;
             this.options = options.stream().toArray(OpenOption[]::new);
-            this.acc = acc;
-            this.filePermissions = filePermissions == null || filePermissions[0] == null
-                            ? EMPTY_FILE_PERMISSIONS : filePermissions;
-            this.isDefaultFS = isDefaultFS(file);
         }
 
         private static boolean isDefaultFS(Path file) {
```

Comment:
Is this method used anymore or can it be removed?

Reply:
Good point. Removed.